python/servo/command_base.py

```diff
@@ -33,6 +35,54 @@ def cd(new_path):
         os.chdir(previous_path)
 
 
+@contextlib.contextmanager
+def setlocale(name):
+    """Context manager for changing the current locale"""
+    saved_locale = locale.setlocale(locale.LC_ALL)
+    try:
+        yield locale.setlocale(locale.LC_ALL, name)
+    finally:
+        locale.setlocale(locale.LC_ALL, saved_locale)
+
+
+def archive_deterministically(dir_to_package, dest_archive, prepend_path=None):
+    """Create a .tar.gz package in a deterministic (reproducible) manner.
+
+    See https://reproducible-builds.org/docs/archives/ for more details. """
+
+    def reset(tarinfo):
+        """Helper to reset owner/group and modification time for tar entries"""
+        tarinfo.uid = tarinfo.gid = 0
+        tarinfo.uname = tarinfo.gname = "root"
+        tarinfo.mtime = 0
+        return tarinfo
+
+    with cd(dir_to_package):
```

Comment:
How hard would it be to get rid of this `with cd(...)` block? I've never been a fan of `cd`ing during builds, but it's OK to keep if it's complicated otherwise.